### DISCUSSION TITLE
One of our sites broke

### DISCUSSION THREAD

**Discussion by totoweb** [akeeba/panopticon#783] [Q&A (Community Support)]
Hi,

One of our sites broke after the update of Panopticon. This is the first error.

`AH01071: Got error 'PHP message: PHP Fatal error: Uncaught Error: Class "ComposerAutoloaderInitb25e220b440613ef9279a3625e03eb71" not found in [/var/www/vhosts/yoga-minds.nl/httpdocs/libraries/vendor/autoload.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Flibraries%2Fvendor&subscriptionId=9&file=autoload.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9):25\nStack trace:\n#0 [/var/www/vhosts/yoga-minds.nl/httpdocs/libraries/bootstrap.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Flibraries&subscriptionId=9&file=bootstrap.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9)(42): require()\n#1 [/var/www/vhosts/yoga-minds.nl/httpdocs/includes/framework.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Fincludes&subscriptionId=9&file=framework.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9)(17): require_once('...')\n#2 [/var/www/vhosts/yoga-minds.nl/httpdocs/includes/app.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Fincludes&subscriptionId=9&file=app.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9)(29): require_once('...')\n#3 [/var/www/vhosts/yoga-minds.nl/httpdocs/index.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs&subscriptionId=9&file=index.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9)(32): require_once('...')\n#4 {main}\n thrown in [/var/www/vhosts/yoga-minds.nl/httpdocs/libraries/vendor/autoload.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Flibraries%2Fvendor&subscriptionId=9&file=autoload.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9) on line 25', referer: https://yoga-minds.nl/`

I see this warning on various previous times;

`AH01071: Got error 'PHP message: PHP Warning: is_dir(): open_basedir restriction in effect. File(/var/lib/mysql/) is not within the allowed path(s): (/var/www/vhosts/yoga-minds.nl/:/tmp/) in [/var/www/vhosts/yoga-minds.nl/httpdocs/api/components/com_panopticon/src/Library/ServerInfo.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Fapi%2Fcomponents%2Fcom_panopticon%2Fsrc%2FLibrary&subscriptionId=9&file=ServerInfo.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9) on line 715; PHP message: PHP Warning: file_exists(): open_basedir restriction in effect. File(/usr/local/etc/os-release) is not within the allowed path(s): (/var/www/vhosts/yoga-minds.nl/:/tmp/) in [/var/www/vhosts/yoga-minds.nl/httpdocs/api/components/com_panopticon/src/Library/ServerInfo.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Fapi%2Fcomponents%2Fcom_panopticon%2Fsrc%2FLibrary&subscriptionId=9&file=ServerInfo.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9) on line 608; PHP message: PHP Warning: file_exists(): open_basedir restriction in effect. File(/etc/os-release) is not within the allowed path(s): (/var/www/vhosts/yoga-minds.nl/:/tmp/) in [/var/www/vhosts/yoga-minds.nl/httpdocs/api/components/com_panopticon/src/Library/ServerInfo.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Fapi%2Fcomponents%2Fcom_panopticon%2Fsrc%2FLibrary&subscriptionId=9&file=ServerInfo.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9) on line 608; PHP message: PHP Warning: file_exists(): open_basedir restriction in effect. File(/proc/stat) is not within the allowed path(s): (/var/www/vhosts/yoga-minds.nl/:/tmp/) in [/var/www/vhosts/yoga-minds.nl/httpdocs/api/components/com_panopticon/src/Library/ServerInfo.php](https://ttwb5.2tw.nl:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Fapi%2Fcomponents%2Fcom_panopticon%2Fsrc%2FLibrary&subscriptionId=9&file=ServerInfo.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F9) on line 330'`

**Comment by nikosdion** [comment]
The problem is that your server has PHP's OPcode cache without giving any software running on your server – such as Panopticon or Joomla! itself – the ability to reset it. As a result, Joomla! got updated, the randomly generated Composer autoloader class name changed, and your site is broken. This has absolutely nothing to do with Panopticon, or even Joomla! itself. It's all because of what your host is doing on their server.

Ask your host to reset the PHP opcode cache.

**Comment by totoweb** [comment]
Ok, but now I see on different servers,Plesk and Direct admin that they say that Joomla 5.1.2 is up to date. They all should have been updated by Panopticon last night. Why does Joomla not see the update anymore?
<img width="453" alt="Scherm­afbeelding 2024-08-21 om 17 23 40" src="https://github.com/user-attachments/assets/790614be-65dc-437d-821a-da6946cc4c30">

**Reply by nikosdion** [reply]
Please read the documentation: https://github.com/akeeba/panopticon/wiki/Periodic-information-update#the-refreshsiteinfo-task There is a reason I am asking. It will provide context to what I am telling you here.

As noted in the documentation, Akeeba Panopticon does not determine by itself the updates available for your site. Joomla! itself, running on your site, is solely responsible for determining whether there is an update available, and which version it is. Panopticon can only ask Joomla for this information.

Since your site reports no update is available, that's what Panopticon knows about your site. Therefore, it should not be trying to install an update it does not know about. That's exactly what happened.

Your _actual_ problem is that Joomla! itself does not detect updates to itself.

As it happens – and despite the obvious fact that we have absolutely no control over Joomla's update process whatsoever – I have a good idea of what is going on, merely because my sites were affected as well and I spent last night and this morning troubleshooting this issue and creating a workaround.

It appears the whole problem comes from Joomla somehow caching wrong update metadata in the `#__tuf_metadata` database table. Due to the nature of TUF (“secure updates”, as Joomla consistently misrepresents them), the problematic metadata either prevent you from seeing new updates, or they cause Joomla to report the _wrong_ updates (in one of my sites the unstable testing release for the upcoming Joomla! 5.2.0 was reported as an available update despite being on the Default update channel, and having minimum stability set to Stable).

This is not an easy issue to fix manually, again because of the nature of TUF updates and the fact that this feature was grafted on top of Joomla's very idiosyncratic updater. Luckily, I happen to understand TUF, Joomla's implementation of TUF, and Joomla's updater.

![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExdXVubzAzN2JnMTM4MTUzYWlrMXN1NWY3dDhjcTZmOW5nYWh5cTV3biZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/UpWDPgxcHiR1e/giphy.gif)

Since I am not allowed to contribute code back to Joomla! anymore I created a new Admin Tools feature to work around broken updates in Joomla! 5 – and also tackle some conditions under which updates could break in Joomla! 4, because why the hell not?

If you are an Admin Tools Professional subscriber you can find a development release for Admin Tools Professional [here](https://www.akeeba.com/download/admin-tools-professional/7-6-0-dev202408211604-revad0fb347.html). If you are not a subscriber, you can find a development release for the free of charge Admin Tools Core [here](https://www.akeeba.com/download/admin-tools-core/7-6-0-dev202408211604-revad0fb347.html). 

Download and install this development release, then go to your site's administrator, Components, Admin Tools for Joomla!, Control Panel. Scroll down and click on the Reset Joomla! Update button. In the new page, click the big Reset Joomla! Update button; it only takes a few moments to come back to the same page. Now go to System, Update, Joomla. You should see the available update. 

If you do not, ask your host if they are caching outbound HTTPS requests, and how you can clear that cache so your site can be updated. Once the cache is cleared, go again to System, Update, Joomla and click on Check for Updates.
